src/app/userManagement/components/register/register.component.ts

```diff
@@ -0,0 +1,52 @@
+import { Component, inject } from '@angular/core';
+import { FormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { Router, RouterLink, RouterOutlet } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { registerUser } from '../../store/actions';
+import { AppState } from '../../../models/state';
+import { Observable } from 'rxjs';
+import { IUser } from '../../models/user';
+import { selectUser } from '../../store/selectors';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  standalone: true,
+  imports: [ReactiveFormsModule, RouterLink, RouterOutlet],
+})
+export class RegisterComponent {
+  fb = inject(FormBuilder);
+  router = inject(Router);
+  store: Store<AppState> = inject(Store);
+  user$: Observable<IUser | undefined> = this.store.select(selectUser);
+
+  hasUnsavedChanges = true;
+
+  form = this.fb.nonNullable.group({
+    email: ['', Validators.required],
+    name: ['', Validators.required],
+    surname: ['', Validators.required],
+    password: ['', Validators.required],
+    confirmPassword: ['', Validators.required],
```

Comment:
You could add patterns here to match the password with regex.

Reply:
Yea i will do so in the Error handing PR, I am just focusing on a happy path for now